torch/_inductor/codegen/cpp.py

```diff
@@ -2244,35 +2171,88 @@ def run(kernel):
 
         scalar_kernel = codegen_kernel(CppKernel)
         inner_most_idx = len(scalar_kernel.itervars) - 1
-        self.call_ranges = scalar_kernel.call_ranges
+        self.set_ranges(group, reduction_group)
         self.loop_nest = LoopNestWithSplit.build(scalar_kernel)
 
         if not self.picked_vec_isa:
             return
 
-        # TODO(jgong5): support alternative tiling factors and data types
-        tiling_factor = self.picked_vec_isa.nelements(dtype=torch.float)
+        def select_tiling_indices():
+            all_index = []
+            for node in nodes:
+                rw = dependencies.extract_read_writes(node._body, *node._sizes)
+                all_index += [dep.index for dep in itertools.chain(rw.reads, rw.writes)]
+            contig_vars = set()
+            non_contig_stride_const = set()
+            non_contig_stride_other = set()
+            for index in all_index:
+                for var in index.free_symbols:
+                    if "tmp" in var.name:
+                        return []
+                    stride = self.stride_at(var, index)
+                    if stride == 1:
+                        contig_vars.add(int(var.name[1:]))
```

Comment:
Variable names can sometimes be stuff like `indirect0` so I don't think converting vars to ints will work in general.  Suggest adding sort function based on something like:
```py
def sort_key(x):
   return (*[int(m) for m in re.findall(r"\d+", x)], x)
```

Reply:
I intended only to check the loop variables and get the indices from them. So I revised the logic to make sure they are loop variables. Please check.